### PR TITLE
RoR Group Capstone - Recipe App

### PR DESCRIPTION
**Pull request by Enning94**
### Recipe_App: Review
In this project, we:
- Followed the layout of the wireframes provided.
- Built login page and registration page with Devise.
- Created Food list:
   -  It displays a list of food and the quantity of food added by the logged-in user.
   - It leads to a form that allows users to add new food.
- Created Recipes list:
    - It displays a list of recipes created by the logged-in user.
    - It leads to recipe details.
    - If the user is the owner of the recipe, it allows the user to delete it.
- Created Public recipe list:
  - It displays a list of all public recipes ordered by the newest.
  - It leads to recipe details.
- Created Recipe details:
  - It displays a toggle button that allows for a recipe to be made public or private.
  - If the recipe is public or the user is the owner of the recipe, it displays the recipe details.
  - If the user is the owner of the recipe, it leads to the form that allows the user to add new food.
- Created General shopping list view :
  - It shows the list of food that is missing for all recipes of the logged-in user
  - It counts the total number of food items and the total price of the missing food.
- Created a navigation menu that allows users to open all of the pages we created.
- Wrote unit and integration tests

### CODE REVIEW

File: spec/models/recipe_spec.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    expect(recipe).to_not be_valid
+  end
+
+  it 'is valid with a name between 3 and 50 characters' do
+    recipe.name = 'a' * 10
+    expect(recipe).to be_valid
+  end
+
+  it 'is not valid with name that already exists' do
+    recipe2 = Recipe.create(user_id: 1, name: 'Recipe 1', description: 'Recipe 1 description', cooking_time: 1,
```
Comment (joskalenda) [inline spec/models/recipe_spec.rb:37]: - [ ] This line is causing the test to fail, as shown in the image below, I would appreciate you fix this issue, and kudos on your job, you did an amazing job

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @Enning94  and @Odongo006 , |
|Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there! |
|  |

## Highlights

- [ ].. ✅ Green linters
- [ ].. ✅ App is working correctly
- [ ].. ✅ Great readme file

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/recipe_spec.rb (hunk 1 above)
Comment (Enning94) [reply spec/models/recipe_spec.rb:37]: Screenshot 2023-09-15 at 10 43 13 PM


### Attached is a screenshot showing all tests are passing locally, we will appreciate further description of what is causing the error in different environment and how we can fix that. Thank you

File: README.md [hunk 2]
```diff
@@ -1 +1,167 @@
...
+- I would like to thank Microverse for giving me the opportunity to work in this project.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
+
```
Comment (joel-kalema) [inline README.md:166]: - [ ] [OPTINAL] Please fix the LICENCE link to make your README more professional 👍 

Screenshot 2023-09-16 123319

**Review by joel-kalema — APPROVED** [review]
Hi Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
Tests are passing 👍 
Screenshot 2023-09-16 131303

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (Enning94) [reply README.md:166]: License fixed, thanks

**Pull request merged**